### PR TITLE
Add parameter that auto-refreshes the window.

### PR DESCRIPTION
**Pull request by MartyLake**
The paramater is called refresh_every.

### CODE REVIEW

File: taskswamp
Comment (danielshahaf) [inline taskswamp]: Please document the new config key.  (Note that sleep(1) supports fractional arguments on some system, but POSIX only requires it to support integers.)

As written this is going to result in a ValueError exception if the config value is set and is not `int()`-able.  Should we try for a nicer error message?

If the value is set to a negative number the user will get an assertion error.  Could you add a message in that case, like "Invalid configuration" in the previous hunk?

Comment (danielshahaf) [inline taskswamp]: Secondarily, more as a matter of coding style: it would be more idiomatic to write `refresh_every = config.get('refresh_every', 0)`, i.e., to use `.get()` and to keep the value as a numeric type internally.

File: bin/forever
Comment (danielshahaf) [inline bin/forever]: The comments here don't look right.  There should be a comment at the top explaining the script's signature (what it expects its arguments to be) and semantics.

**Review by danielshahaf — CHANGES_REQUESTED** [review]
Looks good overall, please see the review comments for details.

File: bin/forever
Comment (MartyLake) [reply bin/forever]: I tried to be more explicit in 502e622, does it look better ? :)

File: taskswamp
Comment (MartyLake) [reply taskswamp]: I documented the new key in the readme in 9c3b9375e54

Comment (MartyLake) [reply taskswamp]: Gentler error handling is in 73003bb

Comment (MartyLake) [reply taskswamp]: fixed in 2e982473

**Comment by MartyLake** [comment]
@danielshahaf I implemented all your suggested changes. Please review again, thanks :)

File: README.md
Comment (danielshahaf) [inline README.md]: "every specified seconds" isn't grammatical, is it?  One alternative would be "set to a value `N` greater than 0" and "every `N` seconds" (but feel free to rephrase if needed).

File: bin/forever
Comment (danielshahaf) [inline bin/forever]: Typo "frist".

Suggest to add "repeatedly" to the first sentence to clarify that it's multiple invocations.

**Review by danielshahaf — COMMENTED** [review]
Looks much better now, thanks!  See comments.

**Comment by MartyLake** [comment]
@danielshahaf sorry for the typos. should be fixed by now :)

**Comment by MartyLake** [comment]
Should be fixed by now, please review, thanks :)

File: bin/forever
Comment (danielshahaf) [inline bin/forever]: Did you test the refresh_every=0 case?  On my system, with the config file from README.md, $1 is `0.0` so this condition evaluates to false, then line 19 is executed as a busy-wait loop.

```sh
% sh -c '[ 0.0 -eq 0 ]'
zsh:[:1: integer expression expected: 0.0
zsh: exit 2     zsh -c '[ 0.0 -eq 0 ]'
```

**Review by danielshahaf — CHANGES_REQUESTED** [review]
I'm afraid the refresh_every=0 case is broken :(

**Comment by danielshahaf** [comment]
I'd be happy to merge once that's fixed. :-)

**Comment by MartyLake** [comment]
That's right! Should be fixed by now :P

**Comment by danielshahaf** [comment]
Sorry I took so long to get back to you, I was busy with other hats.

I've now tweaked your fix in 36e1bd1 to a more readable/idiomatic version, squashed merged, and pushed in 68a35ca.  Thanks again for the patch and sorry again that I dropped the ball.

**Comment by danielshahaf** [comment]
This seems like as good a time as any to say this: I don't use taskswamp nowadays, so I'm looking for someone to take over the maintainership responsibilities.  (I've set up a @taskswamp github organization to move the repository to.)

**Comment by MartyLake** [comment]
Thanks you for merging this repo ! It was interesting contributing to it, the use case I imagined for it is not really what I'm using at the moment. I've never maintainted anything though, it would be interesting to do so !

**Comment by danielshahaf** [comment]
You'd be most welcome. Just subscribe to the repo and start answering PR's & issues as they come in. :-)

**Comment by MartyLake** [comment]
Please add me to the org then :)


---
Matthieu TALBOT
neogaldr@gmail.com

+33 629 22 15 70
fr.linkedin.com/in/matthieutalbot
http://www.soundcloud.com/Marty-Lake
http://lifehacker.com/5028808/how-eom-makes-your-email-more-efficient

**Comment by danielshahaf** [comment]
Will do once I'm back at my workstation. Welcome to the team!

**Comment by danielshahaf** [comment]
@MartyLake I've migrated the repository to https://github.com/taskswamp (from /danielshahaf) and invited you.  For your first task as new maintainer you could update any self-links in the documentation and ask Taskwarrior upstream to update taskswamp's https://taskwarrior.org/tools/ entry.  Happy maintainership!

(I'll stay subscribed, but best highlight me — i.e., mention my nick, @danielshahaf or "danielsh" — if you need my attention.)

**Pull request closed without merging**
